components/Sing.tsx

```diff
@@ -31,25 +34,44 @@ async function searchYoutube(query: string): Promise<YoutubeResult[]> {
 
   const nextRawResp = await fetch("https://www.googleapis.com/youtube/v3/videos?" + nextParams);
   const nextResp = await nextRawResp.json();
-
   return nextResp.items?.map((item: any) => ({
     title: item.snippet.title,
-    thumbnailUrl: item.snippet.thumbnails?.default.url
+    thumbnailUrl: item.snippet.thumbnails?.default.url,
+    videoId: item.id
   })) ?? [];
 }
 
 const Sing = (): React.ReactElement => {
   const [songs, setSongs] = useState<YoutubeResult[]>([])
   const [query, setQuery] = useState('');
-  const [queue, setQueue] = useState([]);
+  const [queue, setQueue] = useState<QueueEntry[]>([]);
+  const [username, setUsername] = useState("");
+  const [showAddModal, setShowAddModal] = useState(false);
+  const [chosenSong, setChosenSong] = useState<YoutubeResult>({
+      title: "",
+      thumbnailUrl: "",
+      videoId: ""
+  })
 
   const router = useRouter();
-  const joinCode = "asdf";
+  const joinCode: any = router.query.joinCode;
 
-  function handleChange(e: any) {
+  function handleQueryChange(e: any) {
     setQuery(e.target.value);
   }
 
+  function handleUsernameChange(e: any) {
+    setUsername(e.target.value);
+  }
+
+  function getNextAvailableIdString(){
```

Comment:
https://www.npmjs.com/package/uuid would be a good option for this. This one will probably cause conflicts that result in dataloss